### ISSUE TITLE
fwts coludn't display time of resume correctly.

### ISSUE THREAD

**Issue by hanhsuan** [canonical/checkbox#380]
### Bug Description

According to [fwts](https://git.launchpad.net/fwts/tree/src/acpi/s3/s3.c?id=f0b577c20af0e24bf5f843d0e41225194c164779#n513), fwts only reports resume time while kernel log printed `Timekeeping suspended`. However, [kernel](https://github.com/torvalds/linux/commit/b3e9431854e8f305385d5de225441c0477b936cb) seems like only to print this message for warning purpose and make fwts to log `Could not determine time to resume`. Finally, this  makes  [commit](https://github.com/hanhsuan/checkbox/commit/803119d075009f4e74df348c5eaadd7994714e24) to report test failed for `id: power-management/suspend-30-cycles-time-check-with-reboots` and `id: suspend/{{ index }}_suspend-single-log-attach_{{ product_slug }}_auto`. For now, it only happens on `AMD Ryzen 5 7530U with Radeon Graphics`.

### To Reproduce

Run job `id: power-management/suspend_30_cycles` and `id: power-management/suspend-30-cycles-time-check-with-reboots`

### Environment

- Ubuntu 22.04
- Kernel: 6.1.0-1007
- Checkbox type: debian
- Checkbox 2.1.0
- Processor:  AMD Ryzen 5 7530U with Radeon Graphics
- [C3 report link of stress](https://certification.canonical.com/hardware/202212-30910/submission/305499/)
- [C3 report link of auto](https://certification.canonical.com/hardware/202212-30910/submission/305393/)

### Relevant log output

```shell
Log from:pm_test.reboot.3.log
================================= Iteration 30 =================================
Results generated by fwts: Version V23.01.00 (2023-01-16 04:09:35).

Some of this work - Copyright (c) 1999 - 2021, Intel Corp. All rights reserved.
Some of this work - Copyright (c) 2010 - 2021, Canonical.
Some of this work - Copyright (c) 2016 - 2021, IBM.
Some of this work - Copyright (c) 2017 - 2021, ARM Ltd.

This test run on 20/03/23 at 09:57:08 on host Linux u-Inspiron-15-3535
6.1.0-1007-oem #7-Ubuntu SMP PREEMPT_DYNAMIC Wed Feb 8 15:41:05 UTC 2023 x86_64.

Command: "fwts -q --stdout-summary -r /var/tmp/checkbox-ng/sessions
/session_title-2023-03-20T08.09.42.session/session-share/fwts.log
--s3-device-check --s3-sleep-delay 30 s3".
Running tests: s3.

s3: Sleep suspend/resume test.
--------------------------------------------------------------------------------
Test 1 of 1: Sleep suspend/resume test.
Defaulted to 1 test, use --s3-multiple=N to run more s2idle cycles 
s2idle cycle 1 of 1 
Detecting the power method.
Response to CanSuspend is yes 
User allowed to execute the CanSuspend action 
Using logind as the default power method.
Cannot open /sys/class/typec to scan Type-C devices.
Requesting Suspend action 
Skipping the minimum delay (0) and using a 3 seconds delay instead 
s2idle duration = 30.
pm-action returned 0 after 30 seconds.
Cannot open /sys/class/typec to scan Type-C devices.
Suspend/Resume Timings:
  Suspend: 0.252 seconds.
  Could not determine time to resume.
Completed s2idle cycle(s) 
PASSED: Test 1, No kernel log errors detected.
PASSED: Test 1, No PM related suspend issues detected.
PASSED: Test 1, No device errors detected.
PASSED: Test 1, No kernel oopses detected.
PASSED: Test 1, No kernel WARN_ON warnings detected.
PASSED: Test 1, No s2idle errors detected.
PASSED: Test 1, Found no errors doing 1 suspend/resume cycle(s).
PASSED: Test 1, All suspends took less than 15.00 seconds.
PASSED: Test 1, All resumes took less than 15.00 seconds.

================================================================================
9 passed, 0 failed, 0 warning, 0 aborted, 0 skipped, 0 info only.
================================================================================

Log from journal log:
Mar 20 09:57:11 u-Inspiron-15-3535 fwts: Starting fwts suspend
Mar 20 09:57:11 u-Inspiron-15-3535 fwts: Suspend
Mar 20 09:57:11 u-Inspiron-15-3535 kernel: wlp2s0: deauthenticating from 24:5a:4c:13:c7:1a by local choice (Reason: 3=DEAUTH_LEAVING)
Mar 20 09:57:11 u-Inspiron-15-3535 kernel: PM: suspend entry (s2idle)
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: Filesystems sync: 0.003 seconds
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: PM: Preparing system for sleep (s2idle)
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: Freezing user space processes ... (elapsed 0.001 seconds) done.
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: OOM killer disabled.
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: Freezing remaining freezable tasks ... (elapsed 0.001 seconds) done.
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: PM: Suspending system (s2idle)
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: printk: Suspending console(s) (use no_console_suspend to debug)
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: PM: suspend of devices complete after 200.405 msecs
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: PM: start suspend of devices complete after 200.562 msecs
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: PM: late suspend of devices complete after 0.639 msecs
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: ACPI: EC: interrupt blocked
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: PM: noirq suspend of devices complete after 42.305 msecs
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: ACPI: \_SB_.PCI0.GPP6.NVME: LPI: Constraint not met; min power state:D3hot current power state:D0
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: PM: suspend-to-idle
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: dell_wmi: Unknown key with type 0x0010 and code 0x0140 pressed
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: dell_wmi: Invalid length of WMI event
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: dell_wmi: Unknown key with type 0x0010 and code 0x0140 pressed
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: dell_wmi: Invalid length of WMI event
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: PM: Triggering wakeup from IRQ 1
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: PM: Triggering wakeup from IRQ 7
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: ACPI: PM: Wakeup unrelated to ACPI SCI
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: PM: resume from suspend-to-idle
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: ACPI: EC: interrupt unblocked
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: dell_wmi: Unknown key with type 0x0010 and code 0x0130 pressed
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: dell_wmi: Invalid length of WMI event
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: PM: noirq resume of devices complete after 28.051 msecs
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: PM: early resume of devices complete after 0.500 msecs
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: pci 0000:00:00.2: can't derive routing for PCI INT A
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: pci 0000:00:00.2: PCI INT A: no GSI
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: [drm] PCIE GART of 1024M enabled.
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: [drm] PTB located at 0x000000F400A00000
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: amdgpu 0000:03:00.0: amdgpu: SMU is resuming...
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: hpet: Lost 1852 RTC interrupts
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: amdgpu 0000:03:00.0: amdgpu: dpm has been disabled
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: amdgpu 0000:03:00.0: amdgpu: SMU is resumed successfully!
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: nvme nvme0: 16/0/0 default/read/poll queues
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: [drm] VCN decode and encode initialized successfully(under DPG Mode).
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: [drm] JPEG decode initialized successfully.
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: amdgpu 0000:03:00.0: amdgpu: ring gfx uses VM inv eng 0 on hub 0
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: amdgpu 0000:03:00.0: amdgpu: ring comp_1.0.0 uses VM inv eng 1 on hub 0
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: amdgpu 0000:03:00.0: amdgpu: ring comp_1.1.0 uses VM inv eng 4 on hub 0
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: amdgpu 0000:03:00.0: amdgpu: ring comp_1.2.0 uses VM inv eng 5 on hub 0
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: amdgpu 0000:03:00.0: amdgpu: ring comp_1.3.0 uses VM inv eng 6 on hub 0
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: amdgpu 0000:03:00.0: amdgpu: ring comp_1.0.1 uses VM inv eng 7 on hub 0
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: amdgpu 0000:03:00.0: amdgpu: ring comp_1.1.1 uses VM inv eng 8 on hub 0
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: amdgpu 0000:03:00.0: amdgpu: ring comp_1.2.1 uses VM inv eng 9 on hub 0
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: amdgpu 0000:03:00.0: amdgpu: ring comp_1.3.1 uses VM inv eng 10 on hub 0
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: amdgpu 0000:03:00.0: amdgpu: ring kiq_2.1.0 uses VM inv eng 11 on hub 0
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: amdgpu 0000:03:00.0: amdgpu: ring sdma0 uses VM inv eng 0 on hub 1
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: amdgpu 0000:03:00.0: amdgpu: ring vcn_dec uses VM inv eng 1 on hub 1
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: amdgpu 0000:03:00.0: amdgpu: ring vcn_enc0 uses VM inv eng 4 on hub 1
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: amdgpu 0000:03:00.0: amdgpu: ring vcn_enc1 uses VM inv eng 5 on hub 1
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: amdgpu 0000:03:00.0: amdgpu: ring jpeg_dec uses VM inv eng 6 on hub 1
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: ata1: SATA link down (SStatus 0 SControl 300)
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: ata2: SATA link down (SStatus 0 SControl 300)
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: usb 3-1: reset full-speed USB device number 2 using xhci_hcd
Mar 20 09:57:41 u-Inspiron-15-3535 kernel: PM: resume of devices complete after 497.081 msecs
```

### Additional context

[journal.log of stress](https://github.com/canonical/checkbox/files/11024694/journal.log)
[pm_test.reboot.3.log of stress](https://github.com/canonical/checkbox/files/11024695/pm_test.reboot.3.log)

**Comment by superm1** [comment]
IMO a better way to capture suspend resume time is to look at a timestamp before suspend starts and to look at timestamp after it ends and compare those.

On AMD systems with s2idle another thing you can do is with dynamic debugging turned on for amd-pmc you can get this message from the kernel 6.3 or less to tell hardware sleep time `Last suspend in deepest state for %lluus`.

In kernel 6.4 there are new dedicated sysfs files that can tell you this time.
https://github.com/torvalds/linux/commit/b52124a78ab34eb0754e32edc0c9996937779176

**Comment by pieqq** [comment]
> IMO a better way to capture suspend resume time is to look at a timestamp before suspend starts and to look at timestamp after it ends and compare those.

Checkbox could implement such a thing in the suspend time measurement test cases, but I feel like this kind of feature should be made available from within FWTS directly. @Ivanhu5866 Do you think it's something doable by FWTS? Should @hanhsuan open a bug in the [FWTS Launchpad project?](https://launchpad.net/fwts)

The challenge is that the solution has to work on a broad variety of Linux kernels, since we still have to test Ubuntu UC20, 18.04, 20.04 and 22.04...

**Comment by superm1** [comment]
My 2 cents:

You should split this into two tests.

The first test should check the suspend duration. This should be done by check-in timestamps before and after test. The purpose for this test is to check for spurious wakeups.
This test should be mandatory.

The second test should look at those new sysfs files I pointed out. The values found should be compared to the duration of the sleep cycle and they should be above some percentage of the time that you can decide.
The purpose for this test is to ensure that the hardware is actually in proper state and not just burning energy.
This test should be optional and only run if the files exist. There are 4 kernel patches for this and they will only export sysfs files if using s2idle. You can backport these to all the kernels you're interested in running this test.

**Comment by Ivanhu5866** [comment]
I think it is doable, but from the kernel log([journal.log of stress](https://github.com/canonical/checkbox/files/11024694/journal.log)
I do see "Timekeeping suspended", it only misses the "Timekeeping suspended" on the last test of suspend.

~~~
Mar 20 09:46:08 u-Inspiron-15-3535 fwts: Starting fwts suspend
Mar 20 09:46:08 u-Inspiron-15-3535 fwts: Suspend
Mar 20 09:46:08 u-Inspiron-15-3535 kernel: wlp2s0: deauthenticating from 24:5a:4c:13:c7:1a by local choice (Reason: 3=DEAUTH_LEAVING)
Mar 20 09:46:09 u-Inspiron-15-3535 kernel: PM: suspend entry (s2idle)
Mar 20 09:46:38 u-Inspiron-15-3535 kernel: Filesystems sync: 0.002 seconds
Mar 20 09:46:38 u-Inspiron-15-3535 kernel: PM: Preparing system for sleep (s2idle)
Mar 20 09:46:38 u-Inspiron-15-3535 kernel: Freezing user space processes ... (elapsed 0.001 seconds) done.
Mar 20 09:46:38 u-Inspiron-15-3535 kernel: OOM killer disabled.
Mar 20 09:46:38 u-Inspiron-15-3535 kernel: Freezing remaining freezable tasks ... (elapsed 0.001 seconds) done.
Mar 20 09:46:38 u-Inspiron-15-3535 kernel: PM: Suspending system (s2idle)
Mar 20 09:46:38 u-Inspiron-15-3535 kernel: printk: Suspending console(s) (use no_console_suspend to debug)
Mar 20 09:46:38 u-Inspiron-15-3535 kernel: PM: suspend of devices complete after 212.120 msecs
Mar 20 09:46:38 u-Inspiron-15-3535 kernel: PM: start suspend of devices complete after 212.275 msecs
Mar 20 09:46:38 u-Inspiron-15-3535 kernel: PM: late suspend of devices complete after 0.512 msecs
Mar 20 09:46:38 u-Inspiron-15-3535 kernel: ACPI: EC: interrupt blocked
Mar 20 09:46:38 u-Inspiron-15-3535 kernel: PM: noirq suspend of devices complete after 41.827 msecs
Mar 20 09:46:38 u-Inspiron-15-3535 kernel: ACPI: \_SB_.PCI0.GPP6.NVME: LPI: Constraint not met; min power state:D3hot current power state:D0
Mar 20 09:46:38 u-Inspiron-15-3535 kernel: PM: suspend-to-idle
Mar 20 09:46:38 u-Inspiron-15-3535 kernel: dell_wmi: Unknown key with type 0x0010 and code 0x0140 pressed
Mar 20 09:46:38 u-Inspiron-15-3535 kernel: dell_wmi: Invalid length of WMI event
Mar 20 09:46:38 u-Inspiron-15-3535 kernel: dell_wmi: Unknown key with type 0x0010 and code 0x0140 pressed
Mar 20 09:46:38 u-Inspiron-15-3535 kernel: dell_wmi: Invalid length of WMI event
Mar 20 09:46:38 u-Inspiron-15-3535 kernel: Timekeeping suspended for 28.290 seconds
Mar 20 09:46:38 u-Inspiron-15-3535 kernel: PM: Triggering wakeup from IRQ 7
Mar 20 09:46:38 u-Inspiron-15-3535 kernel: PM: Triggering wakeup from IRQ 1
Mar 20 09:46:38 u-Inspiron-15-3535 kernel: ACPI: PM: Wakeup unrelated to ACPI SCI
Mar 20 09:46:38 u-Inspiron-15-3535 kernel: PM: resume from suspend-to-idle

~~~

You can file a bug to [FWTS Launchpad project](https://bugs.launchpad.net/fwts), I'll fix the missing string issue.

**Comment by hanhsuan** [comment]
@Ivanhu5866 There is another issue for fwts to parse resume time. In the log below, the `Restarting tasks ... done` is split into two lines and make [fwts](https://github.com/ColinIanKing/fwts/blob/0cbfd011bdfef7c234111d523872a57b6dd3d298/src/acpi/s3/s3.c#L520) couldn't get real `s3_resume_finish`, then make this test failed.
```shell
Oct 17 08:31:28 ubuntu kernel: PM: suspend entry (s2idle)
Oct 17 08:31:28 ubuntu systemd[1]: Starting GRUB failed boot detection...
Oct 17 08:31:28 ubuntu kernel: Filesystems sync: 0.012 seconds
Oct 17 08:31:28 ubuntu kernel: PM: Preparing system for sleep (s2idle)
Oct 17 08:31:32 ubuntu kernel: Freezing user space processes ... (elapsed 0.208 seconds) done.
Oct 17 08:31:32 ubuntu kernel: OOM killer disabled.
Oct 17 08:31:32 ubuntu kernel: Freezing remaining freezable tasks ... (elapsed 0.001 seconds) done.
Oct 17 08:31:32 ubuntu kernel: PM: Suspending system (s2idle)
Oct 17 08:31:32 ubuntu kernel: printk: Suspending console(s) (use no_console_suspend to debug)
Oct 17 08:31:32 ubuntu kernel: serial 00:03: disabled
Oct 17 08:31:32 ubuntu kernel: sd 4:0:0:0: [sda] Synchronizing SCSI cache
Oct 17 08:31:32 ubuntu kernel: sd 4:0:0:0: [sda] Stopping disk
Oct 17 08:31:32 ubuntu kernel: e1000e: EEE TX LPI TIMER: 00000011
Oct 17 08:31:32 ubuntu kernel: pcieport 0000:00:1c.0: AER: Corrected error received: 0000:01:00.0
Oct 17 08:31:32 ubuntu kernel: sdhci-pci 0000:01:00.0: PCIe Bus Error: severity=Corrected, type=Data Link Layer, (Transmitter ID)
Oct 17 08:31:32 ubuntu kernel: sdhci-pci 0000:01:00.0:   device [17a0:9755] error status/mask=00001000/00002000
Oct 17 08:31:32 ubuntu kernel: sdhci-pci 0000:01:00.0:    [12] Timeout
Oct 17 08:31:32 ubuntu kernel: PM: suspend of devices complete after 1170.165 msecs
Oct 17 08:31:32 ubuntu kernel: PM: start suspend of devices complete after 1171.098 msecs
Oct 17 08:31:32 ubuntu kernel: PM: late suspend of devices complete after 15.192 msecs
Oct 17 08:31:32 ubuntu kernel: ACPI: EC: interrupt blocked
Oct 17 08:31:32 ubuntu kernel: PM: Triggering wakeup from IRQ 122
Oct 17 08:31:32 ubuntu kernel: PM: noirq suspend of devices complete after 121.498 msecs
Oct 17 08:31:32 ubuntu kernel: ACPI: \_SB_.PC00.PEG1: LPI: Constraint not met; min power state:D3hot current power state:D0
Oct 17 08:31:32 ubuntu kernel: PM: Triggering wakeup from IRQ 9
Oct 17 08:31:32 ubuntu kernel: PM: suspend-to-idle
Oct 17 08:31:32 ubuntu kernel: ACPI: EC: ACPI EC GPE status set
Oct 17 08:31:32 ubuntu kernel: ACPI: PM: Wakeup after ACPI Notify sync
Oct 17 08:31:32 ubuntu kernel: PM: resume from suspend-to-idle
Oct 17 08:31:32 ubuntu kernel: ACPI: EC: interrupt unblocked
Oct 17 08:31:32 ubuntu kernel: PM: noirq resume of devices complete after 93.064 msecs
Oct 17 08:31:32 ubuntu kernel: pcieport 0000:00:1c.0: AER: Multiple Corrected error received: 0000:01:00.0
Oct 17 08:31:32 ubuntu kernel: pcieport 0000:00:1c.0: PCIe Bus Error: severity=Corrected, type=Data Link Layer, (Transmitter ID)
Oct 17 08:31:32 ubuntu kernel: pcieport 0000:00:1c.0:   device [8086:7ab9] error status/mask=00001000/00002000
Oct 17 08:31:32 ubuntu kernel: pcieport 0000:00:1c.0:    [12] Timeout
Oct 17 08:31:32 ubuntu kernel: sdhci-pci 0000:01:00.0: PCIe Bus Error: severity=Corrected, type=Data Link Layer, (Transmitter ID)
Oct 17 08:31:32 ubuntu kernel: sdhci-pci 0000:01:00.0:   device [17a0:9755] error status/mask=00001000/00002000
Oct 17 08:31:32 ubuntu kernel: sdhci-pci 0000:01:00.0:    [12] Timeout
Oct 17 08:31:32 ubuntu kernel: sdhci-pci 0000:01:00.0: AER:   Error of this Agent is reported first
Oct 17 08:31:32 ubuntu kernel: pcieport 0000:00:1c.0: AER: Corrected error received: 0000:01:00.0
Oct 17 08:31:32 ubuntu kernel: pcieport 0000:00:1c.0: AER: Corrected error received: 0000:01:00.0
Oct 17 08:31:32 ubuntu kernel: PM: early resume of devices complete after 1.891 msecs
Oct 17 08:31:32 ubuntu kernel: sd 4:0:0:0: [sda] Starting disk
Oct 17 08:31:32 ubuntu kernel: pcieport 0000:00:1c.0: AER: Corrected error received: 0000:00:1c.0
Oct 17 08:31:32 ubuntu kernel: pcieport 0000:00:1c.0: PCIe Bus Error: severity=Corrected, type=Data Link Layer, (Transmitter ID)
Oct 17 08:31:32 ubuntu kernel: pcieport 0000:00:1c.0:   device [8086:7ab9] error status/mask=00001000/00002000
Oct 17 08:31:32 ubuntu kernel: pcieport 0000:00:1c.0:    [12] Timeout
Oct 17 08:31:32 ubuntu kernel: pcieport 0000:00:1c.0: AER: Corrected error received: 0000:01:00.0
Oct 17 08:31:32 ubuntu kernel: serial 00:03: activated
Oct 17 08:31:32 ubuntu kernel: sdhci-pci 0000:01:00.0: PCIe Bus Error: severity=Corrected, type=Data Link Layer, (Transmitter ID)
Oct 17 08:31:32 ubuntu kernel: sdhci-pci 0000:01:00.0:   device [17a0:9755] error status/mask=00001000/00002000
Oct 17 08:31:32 ubuntu kernel: sdhci-pci 0000:01:00.0:    [12] Timeout
Oct 17 08:31:32 ubuntu kernel: pcieport 0000:00:1c.0: AER: Corrected error received: 0000:01:00.0
Oct 17 08:31:32 ubuntu kernel: i915 0000:00:02.0: [drm] GuC firmware i915/tgl_guc_70.bin version 70.5.1
Oct 17 08:31:32 ubuntu kernel: i915 0000:00:02.0: [drm] HuC firmware i915/tgl_huc.bin version 7.9.3
Oct 17 08:31:32 ubuntu kernel: i915 0000:00:02.0: [drm] HuC authenticated
Oct 17 08:31:32 ubuntu kernel: i915 0000:00:02.0: [drm] GuC submission disabled
Oct 17 08:31:32 ubuntu kernel: i915 0000:00:02.0: [drm] GuC SLPC disabled
Oct 17 08:31:32 ubuntu kernel: usb 1-14: reset full-speed USB device number 5 using xhci_hcd
Oct 17 08:31:32 ubuntu kernel: ata5: SATA link up 6.0 Gbps (SStatus 133 SControl 300)
Oct 17 08:31:32 ubuntu kernel: ata5.00: configured for UDMA/133
Oct 17 08:31:32 ubuntu kernel: PM: resume of devices complete after 2987.469 msecs
Oct 17 08:31:32 ubuntu kernel: PM: Finishing wakeup.
Oct 17 08:31:32 ubuntu kernel: OOM killer enabled.
Oct 17 08:31:32 ubuntu kernel: Restarting tasks ...
Oct 17 08:31:32 ubuntu kernel: mei_hdcp 0000:00:16.0-b638ab7e-94e2-4ea2-a552-d1c54b627f04: bound 0000:00:02.0 (ops i915_hdcp_component_ops [i915])
Oct 17 08:31:32 ubuntu kernel: usb 1-8: USB disconnect, device number 3
Oct 17 08:31:32 ubuntu kernel: done.
Oct 17 08:31:32 ubuntu kernel: random: crng reseeded on system resumption
Oct 17 08:31:32 ubuntu kernel: mei_pxp 0000:00:16.0-fbf6fcf1-96cf-4e2e-a6a6-1bab8cbe36b1: bound 0000:00:02.0 (ops i915_pxp_tee_component_ops [i915])
Oct 17 08:31:32 ubuntu rtkit-daemon[992]: The canary thread is apparently starving. Taking action.
Oct 17 08:31:32 ubuntu kernel: PM: suspend exit
```

**Comment by Hook25** [comment]
We have decided to clean up our issue tracker to systematize our approach to backlog grooming and planning. If this issue is still relevant to you, please reopen it and tag one of the maintainers or reach out on ~Checkbox on Mattermost. Going forward all issues that have received no activity for more than a year will be marked as 'stale' and closed after one week
